### ISSUE TITLE
HttpClient.SendAsync with HttpMethod.Trace and request content sends a body

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#9023]
https://tools.ietf.org/html/rfc7231 states for TRACE "A client MUST NOT send a message body in a TRACE request," but WinHttpHandler does:

``` C#
using (var client = new HttpClient())
{
    var request = new HttpRequestMessage(HttpMethod.Trace, "http://corefx-net.cloudapp.net/Echo.ashx") { Content = new StringContent("some body") };
    using (HttpResponseMessage response = client.SendAsync(request).GetAwaiter().GetResult())
    {
        Console.WriteLine(response.StatusCode);
    }
}
```

This outputs BadRequest.

**Comment by davidsh** [comment]
Not clear what we should here other than throw an exception on the SendAsync if a request body is present.  This would be different behavior than Desktop which I think permits this.

**Comment by davidsh** [comment]
.NET Framework (Desktop) HTTP stack also allows developers to send a body with any HTTP verb including TRACE.

The server (assuming it is RFC compliant) will reject the HTTP request with "400 Bad Request".

I don't think we should change the client behavior, though.  We would be introducing a behavior change by, presumably, throwing a new exception from the .SendAsync() and associated methods.

**Comment by Jiayili1** [comment]
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L1602
Disabled test

**Comment by karelz** [comment]
cc @steveharter @Priya91 @ianhays

**Comment by davidsh** [comment]
.NET Framework also allows the HttpWebRequest and HttpClient APIs to send a request using 'TRACE' verb and a request body.  The usual response from a server is "400 Bad Request".

We will keep .NET Core matching in behavior to .NET Framework.  While the RFC might talk about expectations of the stack, it is describing the protocol behavior overall between client and server, and not necessarily the behavior expected from a client-side API.

The only fix here for this issue is to adjust the test accordingly.

**Comment by ViktorHofer** [comment]
I'm confused. @Jiayili1 you added the disabled-test label and linked to the test method `SendAsync_SendSameRequestMultipleTimesDirectlyOnHandler_Success` which is not same issue. We are specifically talking about the method `SendAsync_SendRequestUsingNoBodyMethodToEchoServerWithContent_NoBodySent` which can be found here https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs#L1639.

There is no active issue set in code --> it is already commented out. I think this issue should be closed. @davidsh explained why "400 Bad Request" is the correct behavior which also mimics netfx behavior.

**Comment by karelz** [comment]
@Jiayili1 can you please double check that this was a mistake? (Just want to make sure we do not miss things by accident) Thank you!

Thanks @ViktorHofer for clarifying!

**Comment by karelz** [comment]
@ViktorHofer what should I think then about this comment from @davidsh? (he is OOF, back on Monday - we can wait on his comment in the worst case)

> The only fix here for this issue is to adjust the test accordingly.

**Comment by ViktorHofer** [comment]
I commented on exactly that statement. See above or here:

> Which is and was already done (I looked into the history). There is no active issue set in code --> it is already commented out. I think this issue should be closed. @davidsh explained why "400 Bad Request" is the correct behavior which also mimics netfx behavior.

**Comment by karelz** [comment]
I see, in that case, link the PR/commit and we can close it ...
